support/util/resources.go

```diff
@@ -0,0 +1,106 @@
+package util
+
+import (
+	"strings"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+
+	agentv1 "github.com/openshift/cluster-api-provider-agent/api/v1beta1"
+
+	capiaws "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	capiazure "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	capiibmv1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta2"
+	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
+	capiopenstackv1beta1 "sigs.k8s.io/cluster-api-provider-openstack/api/v1beta1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	secretsstorev1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+)
+
+var (
+	BaseResources = []client.Object{
+		&capiv1.Cluster{},
+	}
+
+	AWSResources = []client.Object{
+		&capiaws.AWSCluster{},
+		&hyperv1.AWSEndpointService{},
+	}
+
+	AzureResources = []client.Object{
+		&capiazure.AzureCluster{},
+		&capiazure.AzureClusterIdentity{},
+	}
+
+	ManagedAzure = []client.Object{
+		&secretsstorev1.SecretProviderClass{},
+	}
+
+	IBMCloudResources = []client.Object{
+		&capiibmv1.IBMVPCCluster{},
+	}
+
+	KubevirtResources = []client.Object{
+		&capikubevirt.KubevirtCluster{},
+	}
+
+	AgentResources = []client.Object{
+		&agentv1.AgentCluster{},
+	}
+
+	OpenStackResources = []client.Object{
+		&capiopenstackv1beta1.OpenStackCluster{},
+	}
+)
+
+func GetHostedClusterManagedResources(platformsInstalled string) []client.Object {
+	var managedResources []client.Object
+
+	platformsInstalledList := strings.Split(platformsInstalled, ",")
+	for _, platform := range platformsInstalledList {
+		platform = strings.ToLower(strings.TrimSpace(platform))
+		switch {
+		case strings.EqualFold(platform, string(hyperv1.AWSPlatform)):
+			managedResources = append(managedResources, BaseResources...)
```

Comment:
Why not append BaseResources once before the platform-specific ones?

Reply:
The None platform doesn't have that resource and including it by default was causing the unit test to fail.